azure/durable_functions/tasks/task_all.py

```diff
@@ -37,10 +38,14 @@ def task_all(tasks: List[Task]):
         else:
             complete_time = task.timestamp if complete_time is None \
                 else max([task.timestamp, complete_time])
+        if task._is_played:
+            is_played = True
```

Comment:
It seems like we're declaring the whole `TaskSet` as `is_played` if only one of the tasks are played.  I don't really understand the role of this function, but is the correct behavior, or should we be setting `is_played` to `True` only if *all* the tasks have been played?

Reply:
I think the semantics are a bit unclear here. The way JS does it is that is `isReplayed` flag is True if the last event scheduled has a value is `isPlayed = True`, which isn't necessarily all the robust.

I do think that the alternative you propose is much better, so I'll go for it!

Reply:
I do need some clarification on the following: is there a scenario where all the tasks in a TaskSet somehow do not have the same `isPlayed` value? Would that even make sense? @cgillum

Reply:
I would need to know that in order to write a tester

Reply:
@cgillum I changed the logic such that isReplayed is only true if all constituent tasks are played. Also added some testing

Reply:
@davidmrdavid,

As an example of some code that may result in the mismatched isPlayed value you asked about.

```python
task1 = context.call_activity();
yield task1;
task2 = context.call_activity();
combinedTask = context.task_all(task1, task2)
if context.is_replaying:
   #log statement
yield combinedTask
```

Super contrived, but technically valid, and I imagine there could be scenarios that are not as contrived that hit this.